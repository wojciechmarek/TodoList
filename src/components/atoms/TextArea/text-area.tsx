import { TextAreaProps } from './text-area.interface';
import { TextAreaStyle } from './text-area.styled';

export const TextArea = (props: TextAreaProps) => {
  const { value, placeholder, handleTextAreaChange, ...params } = props;

  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    handleTextAreaChange(e.target.value);
  };

  return (
    <TextAreaStyle
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      {...params}
    />
  );
};
