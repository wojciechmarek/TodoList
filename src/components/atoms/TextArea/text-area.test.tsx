import { TextAreaProps } from './text-area.interface';
import '@testing-library/jest-dom';
import { configure, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextArea } from './text-area';

describe('TextArea', () => {
  let textAreaProps: TextAreaProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    textAreaProps = {
      value: 'hello textarea',
      placeholder: 'placeholder test for textarea',
      handleTextAreaChange: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(<TextArea {...textAreaProps} />);
    expect(baseElement).toBeTruthy();
  });

  describe('when the text area is rendered', () => {
    it('should have the correct value', () => {
      const { getByRole } = render(<TextArea {...textAreaProps} />);
      const textarea = getByRole('textbox');
      expect(textarea).toHaveValue('hello textarea');
    });

    it('should have the correct placeholder', () => {
      const { getByRole } = render(<TextArea {...textAreaProps} />);
      const textarea = getByRole('textbox');
      expect(textarea).toHaveAttribute(
        'placeholder',
        'placeholder test for textarea'
      );
    });
  });

  describe('when the text to the text area is entered', () => {
    it('should call the handleTextAreaChange function when the input changes', async () => {
      const { getByRole } = render(<TextArea {...textAreaProps} />);
      const textarea = getByRole('textbox');
      await userEvent.type(textarea, 'qwerty', { delay: 1 });
      expect(textAreaProps.handleTextAreaChange).toHaveBeenCalled();
    });

    it('should call the handleTextAreaChange function 3 times for 3 characters', async () => {
      const { getByRole } = render(<TextArea {...textAreaProps} />);
      const textarea = getByRole('textbox');
      await userEvent.type(textarea, 'abc', { delay: 1 });
      expect(textAreaProps.handleTextAreaChange).toHaveBeenCalledTimes(3);
    });

    it('should call the handleTextAreaChange function with correct phrase', async () => {
      const { getByRole } = render(<TextArea {...textAreaProps} />);
      const textarea = getByRole('textbox');
      await userEvent.type(textarea, '1', { delay: 1 });
      expect(textAreaProps.handleTextAreaChange).toHaveBeenCalledWith(
        'hello textarea1'
      );
    });
  });
});
