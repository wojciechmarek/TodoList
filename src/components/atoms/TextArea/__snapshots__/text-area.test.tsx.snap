// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should match the snapshot 1`] = `
.emotion-0 {
  border: 1px solid var(--search-input-border-and-icon-bg-color);
  background-color: var(--search-input-bg-color);
  color: var(--text-font-color);
  font-weight: 800;
  padding-left: 1em;
  padding-top: 0.5em;
  border-radius: var(--standard-border-radius);
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
}

.emotion-0::-webkit-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-0::-moz-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-0:-ms-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-0::placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<textarea
  className="emotion-0"
  onChange={[Function]}
  placeholder="placeholder test for textarea"
  value="hello textarea"
/>
`;
